13-collabs.Rmd

```diff
@@ -5,6 +5,248 @@
+
+An effective code review atmosphere is something that individuals and their team have to _commit_ to (pun intended). Effective code review brings so many benefits not only to your project quality but also your communication skills through fostering a learning atmosphere!
+
+In this chapter we will discuss the two sides of code review. Code review ideally includes at least two people: the author of the pull request and the reviewer of the pull request. Depending on your job context, we realize that sometimes authors have to become their own reviewers if code review is not something that can be prioritized by your institution or team.
+
+## Author responsibilities in code review
+
+The code review process begins with the creation of a pull request (which we practiced in the previous chapter). Successful and efficient code review is born out of quality communication, which is a skill set on its own. You can set up your reviewers (and yourself) up for success by knowing what basic information can help get the code review conversation going.
+
+```{r, fig.alt="Ruby has filed a very large pull request with a very short PR description:  I updated the project and added new files. She then requested Avi to review this PR. This leaves Avi with very little information to go on and he is very confused. Avi comments: Ummm… I’m a bit confused. Can you explain the context of these changes? This is a lot for me to try to follow.  I’m also unsure what kind of feedback you are looking for. ", out.width = "100%", echo = FALSE}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g33bf0789107_300_2369")
+```
+
+Even if you end up being the only person who will review your own code, writing these things out is still very helpful and highly recommended. It can help you spot problems you might not have otherwise seen and generally help you document better for future you!
+
+## Characteristics of great pull requests and issues
+
+### There's plenty of context!
+
+What's the story behind around the changes you are proposing? Sometimes when we are in the thick of a project we can make the mistake of assuming everyone knows what we know. This can unfortunately leave a huge burden on your reviewer to try to follow a paper trail to try to understand what you are doing.
+
+Before sending off a review request, re-read your PR description and think about the perspective of your reviewer. Err on the side that they have no idea what is happening on the project (because sometimes this is the case!)
+
+Tell a short story to explain what lead to you making these changes including attempting to answer these questions:  
+
+- What is the problem that these changes will solve?
+- Do you have any URLs relevant issues or files you can share?
+- What inspired you to take this approach -- are there other things you tried?
+- Are there other pull requests related to this change?
+
```

Comment:
```suggestion


:::note
You can type the world "resolves" and then include the issue number with a `#` and it will close the issue when you merge the pull request
:::
```

Reply:
thought this might be a nice tip :)

Reply:
Oh perfect! I wanted to add this but didn't want to interrupt the flow but a box is perfect.